sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -81,12 +81,20 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   /**
    * Runs this query returning the result as an RDD.
    */
-  def execute(): RDD[Row]
+  final def execute(): RDD[Row] = {
+    RDDOperationScope.withScope(sparkContext, nodeName, false, true) {
+      doExecute()
+    }
+  }
 
   /**
-   * Runs this query returning the result as an array.
+   * Runs this query returning the result as an RDD.
    */
+  protected def doExecute(): RDD[Row]
 
+  /**
+   * Runs this query returning the result as an array.
+   */
```

Comment:
Can you clarify in the scala doc here the purpose of the two methods.

`execute()`: Returns the result of this query as an `RDD[Row]`.  This function is also responsible for adding query plan information to created RDDs for visualization.  Concrete implementations of `SparkPlan` should override `doExecute`.

`doExecute()`: Overridden by concrete implementations of SparkPlan.  Produces the result of the query as an `RDD[Row]`